builtin/credential/radius/path_config.go

```diff
@@ -0,0 +1,224 @@
+				Description: "radius port (default: 1812)",
+			},
+			"secret": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "secret shared with the radius server",
+			},
+			"allow_unknown_users": &framework.FieldSchema{
+				Type:        framework.TypeBool,
+				Default:     false,
+				Description: "enable granting default policy upon successful RADIUS authentication (default: false)",
+			},
+			"reauth_on_renew": &framework.FieldSchema{
+				Type:        framework.TypeBool,
+				Default:     false,
+				Description: "attempt reauthentication with backend before granting token renewal (default: false)",
+			},
+			"dial_timeout": &framework.FieldSchema{
+				Type:        framework.TypeDurationSecond,
+				Default:     10,
+				Description: "number of seconds before connect timeouts (default: 10)",
+			},
+			"read_timeout": &framework.FieldSchema{
+				Type:        framework.TypeDurationSecond,
+				Default:     10,
+				Description: "number of seconds before response timeouts (default: 10)",
+			},
+			"nas_port": &framework.FieldSchema{
+				Type:        framework.TypeInt,
+				Default:     10,
+				Description: "RADIUS NAS port field (default: 10)",
+			},
+		},
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.ReadOperation:   b.pathConfigRead,
+			logical.UpdateOperation: b.pathConfigWrite,
+		},
+
+		HelpSynopsis:    pathConfigHelpSyn,
+		HelpDescription: pathConfigHelpDesc,
+	}
+}
+
+/*
+ * Construct ConfigEntry struct using stored configuration.
+ */
+func (b *backend) Config(req *logical.Request) (*ConfigEntry, error) {
+	// Schema for ConfigEntry
+	fd, err := b.getConfigFieldData()
+	if err != nil {
+		return nil, err
+	}
+
+	// Create a new ConfigEntry, filling in defaults where appropriate
+	result, err := b.newConfigEntry(fd)
+	if err != nil {
+		return nil, err
+	}
+
+	storedConfig, err := req.Storage.Get("config")
+	if err != nil {
+		return nil, err
+	}
+
+	if storedConfig == nil {
+		// No user overrides, return default configuration
+		return result, nil
+	}
+
+	// Deserialize stored configuration.
+	// Fields not specified in storedConfig will retain their defaults.
+	if err := storedConfig.DecodeJSON(&result); err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
+
+func (b *backend) pathConfigRead(
+	req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+
+	cfg, err := b.Config(req)
+	if err != nil {
+		return nil, err
+	}
+	if cfg == nil {
+		return nil, nil
+	}
+
+	resp := &logical.Response{
+		Data: structs.New(cfg).Map(),
+	}
+	resp.AddWarning("Read access to this endpoint should be controlled via ACLs as it will return the configuration information as-is, including any passwords.")
+	return resp, nil
+}
+
+/*
+ * Creates and initializes a ConfigEntry object with its default values,
+ * as specified by the passed schema.
+ */
+func (b *backend) newConfigEntry(d *framework.FieldData) (*ConfigEntry, error) {
+	cfg := new(ConfigEntry)
```

Comment:
Increasingly growing number of paths in Vault are distinguishing between `create` and `update` capabilities. The `d.Get()` function returns the `default` values set in the corresponding field data. We use it only during `create` operation. During an update, we take in values that are supplied and not refer the default values of the rest. This function should follow that same procedure. Otherwise, its hard to reason about the state of the configuration after multiple writes. I'll show an example that could be replicated to all other fields:

```
	hostRaw, ok := data.GetOk("host")
	if ok {
		cfg.Host = strings.ToLower(hostRaw.(string))
	} else if req.Operation == logical.CreateOperation {
		cfg.Host = strings.ToLower(d.Get("host").(string))
	}

```